File: java/src/main/java/com/gmail/fuskerr63/java/interactor/DatabaseInteractor.java
package com.gmail.fuskerr63.java.interactor;

import com.gmail.fuskerr63.java.entity.ContactLocation;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface DatabaseInteractor {
    Single<List<ContactLocation>> getAll();
    Single<ContactLocation> getUserByContactId(int contactId);
    Completable insert(ContactLocation contactLocation);
}

File: java/src/main/java/com/gmail/fuskerr63/java/interactor/DatabaseInteractor.kt
package com.gmail.fuskerr63.java.interactor

import com.gmail.fuskerr63.java.entity.ContactLocation
import io.reactivex.Completable
import io.reactivex.Single
import kotlinx.coroutines.flow.Flow

interface DatabaseInteractor {
    fun getAll(): Single<List<ContactLocation>>
    fun getFlowUserById(contactId: String): Flow<ContactLocation?>
    fun getSingleUserById(contactId: String): Single<ContactLocation>
    fun insert(contactLocation: ContactLocation): Completable
}
